lib/editor/components/timetable/TimetableGrid.js

```diff
@@ -378,27 +378,35 @@ export default class TimetableGrid extends Component<Props> {
     } = this.props
     let activeRow = rowIndex
     let activeCol = colIndex
-    // iterate over rows in pasted selection
+    // Iterate over rows in pasted selection
     for (var i = 0; i < pastedRows.length; i++) {
       activeRow = rowIndex + i
-      // construct new row if it doesn't exist
-      if (typeof data[i + rowIndex] === 'undefined') {
+      // Construct new row if it doesn't exist
+      if (typeof data[activeRow] === 'undefined') {
         addNewRow()
       }
-      // iterate over number of columns in pasted selection
-      for (var j = 0; j < pastedRows[0].length; j++) {
+      // Iterate over number of columns in pasted selection
+      for (var j = 0; j < pastedRows[i].length; j++) {
```

Comment:
You can keep this as `0` bc the length should remain constant

Reply:
I don't quite remember why I changed that -- thanks for the catch